server/src-lib/Hasura/RQL/Types/ComputedField.hs

```diff
@@ -44,6 +44,17 @@ instance ToJSON FunctionTableArgument where
   toJSON FTAFirst             = String "first_argument"
   toJSON (FTANamed argName _) = object ["name" .= argName]
 
+-- | The session argument, which passes Hasura session variables to a
+-- SQL function as a JSON object.
+data FunctionSessionArgument
+  = FunctionSessionArgument !FunctionArgName !Int
+  deriving (Show, Eq, Generic)
+instance Cacheable FunctionSessionArgument
+
+instance ToJSON FunctionSessionArgument where
+  -- TODO is this the right JSON to output?
+  toJSON (FunctionSessionArgument argName _) = object ["name" .= argName]
```

Comment:
I'm sort of missed this `TODO`. So, the `FunctionSessionArgument` carries the session argument name and it's ordinal position in the function input params. It is enough to just output the name directly without any nesting.
```suggestion
instance ToJSON FunctionSessionArgument where
  toJSON (FunctionSessionArgument argName _) = toJSON argName
```

Also, document the type `FunctionSessionArgument` like the following:
```haskell
= FunctionSessionArgument 
     !FunctionArgName -- ^ The argument name
     !Int -- ^ The ordinal position in the function input parameters
```